core/src/main/java/com/sequenceiq/cloudbreak/service/DatabaseBackupRestoreService.java

```diff
@@ -0,0 +1,83 @@
+package com.sequenceiq.cloudbreak.service;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.dto.NameOrCrn;
+import com.sequenceiq.cloudbreak.controller.validation.dr.BackupRestoreV4RequestValidator;
+import com.sequenceiq.cloudbreak.core.flow2.service.ReactorFlowManager;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.event.ResourceEvent;
+import com.sequenceiq.cloudbreak.exception.BadRequestException;
+import com.sequenceiq.cloudbreak.service.stack.StackService;
+import com.sequenceiq.cloudbreak.structuredevent.event.CloudbreakEventService;
+import com.sequenceiq.cloudbreak.validation.ValidationResult;
+import com.sequenceiq.flow.api.model.FlowIdentifier;
+import com.sequenceiq.flow.core.FlowLogService;
+import com.sequenceiq.flow.domain.FlowLog;
+import java.util.List;
+import javax.inject.Inject;
+import org.apache.commons.collections.CollectionUtils;
+import org.springframework.stereotype.Service;
+
+@Service
+public class DatabaseBackupRestoreService {
+
+    private static final String DATALAKE_DATABASE_BACKUP = "DATALAKE_DATABASE_BACKUP";
+
+    private static final String DATALAKE_DATABASE_RESTORE = "DATALAKE_DATABASE_RESTORE";
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private FlowLogService flowLogService;
+
+    @Inject
+    private ReactorFlowManager flowManager;
+
+    @Inject
+    private CloudbreakEventService eventService;
+
+    @Inject
+    private BackupRestoreV4RequestValidator requestValidator;
+
+    public void validate(Long workspaceId, NameOrCrn nameOrCrn, String location, String backupId) {
+        Stack stack = stackService.getByNameOrCrnInWorkspace(nameOrCrn, workspaceId);
+        ValidationResult validationResult = requestValidator.validate(stack, location, backupId);
+        if (validationResult.hasError()) {
+            throw new BadRequestException(validationResult.getFormattedErrors());
+        }
+    }
+
+    public FlowIdentifier backupDatabase(Long workspaceId, NameOrCrn nameOrCrn, String location, String backupId) {
+        Stack stack = stackService.getByNameOrCrnInWorkspace(nameOrCrn, workspaceId);
+        List<FlowLog> flowLogs = flowLogService.findAllByResourceIdAndFinalizedIsFalseOrderByCreatedDesc(stack.getId());
+        if (!CollectionUtils.isEmpty(flowLogs)) {
+            String errorMsg = String.format("Database backup cannot be performed because there is an active flow running: %s",
+                flowLogs.stream().map(FlowLog::toString));
+            eventService.fireCloudbreakEvent(
+                stack.getId(),
+                DATALAKE_DATABASE_BACKUP,
+                ResourceEvent.DATALAKE_DATABASE_BACKUP_COULD_NOT_START,
+                List.of(errorMsg));
+            throw new BadRequestException(errorMsg);
+        } else {
+            return flowManager.triggerDatalakeDatabaseBackup(stack.getId(), location, backupId);
```

Comment:
it would be nice to have a line of log here after MDCContext is build. same for restore method